sdk/typescript/src/builder/Transaction.ts

```diff
@@ -419,11 +438,13 @@ export class Transaction {
       );
 
       objects.forEach((object, i) => {
-        const { id, input } = objectsToResolve[i];
+        const { id, input, normalizedType } = objectsToResolve[i];
         const initialSharedVersion = getSharedObjectInitialVersion(object);
 
         if (initialSharedVersion) {
-          const mutable = true; // Defaulted to True to match current behavior.
+          const mutable =
+            normalizedType != null &&
+            extractMutableReference(normalizedType) != null;
```

Comment:
cc @amnn this is the latest set mutable change

Reply:
Great, thanks for already being on top of this!  qq -- what happens if the same Object ID is required by multiple calls in the transaction, potentially with different type signatures?  I would expect to see logic to de-duplicate by ID, and pick an upperbound type for the transaction input based on the call parameter types (by-value beats by-mutable-reference, which beats by-reference) but I didn't spot that.  Is that scenario even supported?

Reply:
NIce catch, I'll fix that!